src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.hw4.myArrayList;
+
+import java.util.Iterator;
+
+public class SimpleArray<E> implements Simple<E> {
+    private E[] values;
+
+    public SimpleArray() {
+        values = (E[]) new Object[0];
+    }
+
+    @Override
+    public boolean add(E e) {
+        try {
+            E[] temp = values;
+            values = (E[]) new Object[temp.length + 1];
+            System.arraycopy(temp, 0, values, 0, temp.length);
+            values[values.length - 1] = e;
+            return true;
+        } catch (ClassCastException ex) {
```

Comment:
Why you expect ClassCastException here ?

Reply:
an exception  is possible in this operation: values = (E[]) new Object[temp.length + 1];